opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DefaultAlarmService.java

```diff
@@ -114,18 +120,29 @@ public void escalateAlarm(OnmsAlarm alarm, Date now) {
     @Override
     @Transactional
     public void acknowledgeAlarm(OnmsAlarm alarm, Date now) {
-        LOG.info("Acknowledging alarm with id: {}", alarm.getId());
+        LOG.info("Acknowledging alarm with id: {} @ {}", alarm.getId(), now);
         final OnmsAlarm alarmInTrans = alarmDao.get(alarm.getId());
         if (alarmInTrans == null) {
             LOG.warn("Alarm disappeared: {}. Skipping ack.", alarm);
             return;
         }
-        final String previousAckUser = alarmInTrans.getAlarmAckUser();
-        final Date previousAckTime = alarmInTrans.getAlarmAckTime();
-        alarmInTrans.setAlarmAckUser(DEFAULT_USER);
-        alarmInTrans.setAlarmAckTime(now);
-        alarmDao.update(alarmInTrans);
-        alarmEntityNotifier.didAcknowledgeAlarm(alarmInTrans, previousAckUser, previousAckTime);
+        OnmsAcknowledgment ack = new OnmsAcknowledgment(alarm, DEFAULT_USER, now);
```

Comment:
We should use the alarm that was retrieved within the transaction i.e.:
```suggestion
        OnmsAcknowledgment ack = new OnmsAcknowledgment(alarmInTrans, DEFAULT_USER, now);
```